dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/strategy/AbstractTransformStrategy.java

```diff
@@ -0,0 +1,56 @@
+package com.dotmarketing.portlets.contentlet.transform.strategy;
+
+import static com.dotmarketing.portlets.contentlet.transform.strategy.TransformToolbox.privateInternalProperties;
+
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Entry point class for the Strategies hierarchy
+ * @param <T>
+ */
+public abstract class AbstractTransformStrategy<T extends Contentlet> {
+
+    protected final TransformToolbox toolBox;
+
+    AbstractTransformStrategy(final TransformToolbox toolBox) {
+        this.toolBox = toolBox;
+    }
+
+    /**
+     * Any descendant that has a particular way to retrieve it's own concrete type should override this
+     * @param contentlet
+     * @return
+     */
+     T fromContentlet(Contentlet contentlet){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'contentlet' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)